### PR TITLE
Adding support for static website for Storage Account (Post-deploy task)

### PR DESCRIPTION
**Pull request by Dzoukr**
At Farmer CE:

```fsharp
let store = storageAccount {
    name "romanstaticweb"
    static_website "index.html" "error.html"
}
```

on Azure:

image

Also added docs + configuration member for getting *primary endpoint*

```fsharp
store.WebsitePrimaryEndpoint
```

### CODE REVIEW

**Comment by Dzoukr** [comment]
Oh, it also probably closes #180 if approved 😉

**Comment by Dzoukr** [comment]
Hold on, I got even better idea. Will push soon.

**Comment by Dzoukr** [comment]
Ha, now with batch upload to $web folder!

```fsharp
let stor = storageAccount {
    name "romanstaticweb"
    static_website_with_content "index.html" "error.html" @"c:\temp\web\"
}
```

image

and it works 😄 

image

**Comment by isaacabraham** [comment]
Excellentatious. I'll do a quick review - might change the member naming or something - but this is great and will go in the next release. Thanks!

**Comment by Dzoukr** [comment]
Please do! 🙏 Me and naming stuff - it's not working together 😄

File: src/Farmer/Arm/Storage.fs [hunk 1]
```diff
@@ -24,7 +25,24 @@ type StorageAccount =
                apiVersion = "2018-07-01"
                location = this.Location.ArmValue
             |} :> _
-
+    interface IPostDeploy with
+        member this.Run resourceGroupName =
+            match this with
+            | { StaticWebsite = Some (indexDoc, errorDoc, folder); Name = name } ->
+                printfn "Enabling static web site for storage account %s with Index document as %s, Error document as %s" name.Value indexDoc errorDoc 
+                Deploy.Az.enableStaticWebsite name.Value indexDoc errorDoc
```
Comment (isaacabraham) [inline src/Farmer/Arm/Storage.fs:33]: I wonder if this could be enabled via the ARM template itself?

Comment (Dzoukr) [reply src/Farmer/Arm/Storage.fs:33]: I tried to google + export template from existing static website, but no luck. Also found few mentions that it's not supported for ARM out of the box.

Comment (Dzoukr) [reply src/Farmer/Arm/Storage.fs:33]: Got "inspired" 😉 by this post: http://www.frankysnotes.com/2019/03/deploy-automatically-static-website.html

Comment (isaacabraham) [reply src/Farmer/Arm/Storage.fs:33]: Yeah just saw - a dead end.

**Pull request merged**